### PR TITLE
Do not run/stop workflows that are not required

### PR DESCRIPTION
**Pull request by praveenrewar**
#### What this PR does / why we need it:
- Do not run kapp-controller tests for cli changes
- Cancel previous in progress runs of workflows on same branch

### CODE REVIEW

File: .github/workflows/kind-action.yml [hunk 1]
```diff
@@ -3,8 +3,16 @@ name: Kind Cluster E2E tests
 on:
   pull_request:
     types: [opened, reopened, synchronize]
+    paths-ignore:
+    - 'docs/**'
+    - '*.md'
+    - 'cli/**'
   workflow_dispatch:
 
+concurrency:
```
Comment (neil-hickey) [inline .github/workflows/kind-action.yml:12]: oh this is cool, wasn't aware of it. Nice!

File: .github/workflows/upgrade-testing.yml [hunk 2]
```diff
@@ -12,6 +12,10 @@ name: upgrade-testing
       - 'docs/**'
       - '*.md'
 
+concurrency:
```
Comment (neil-hickey) [inline .github/workflows/upgrade-testing.yml:17]: Do we intend for the upgrade tests to run on cli changes? @praveenrewar

**Review by neil-hickey — APPROVED** [review]
One question about whether we need to run the upgrade tests on the cli, other than that, looks good! This will help our PRs go faster, nice!

File: .github/workflows/upgrade-testing.yml (hunk 2 above)
Comment (praveenrewar) [reply .github/workflows/upgrade-testing.yml:17]: Ah, good catch! Let me make the changes here too.

Comment (praveenrewar) [reply .github/workflows/upgrade-testing.yml:17]: Done!

**Comment by praveenrewar** [comment]
> One question about whether we need to run the upgrade tests on the cli

Good catch! We don't need to, I have made the changes accordingly :)

**Pull request merged**